crates/oxc_cli/src/lib.rs

```diff
@@ -38,6 +38,20 @@ impl Cli {
             .cli_options
             .paths
             .iter()
+            .filter(|&path| {
+                let ignore_pattern = &self.cli_options.ignore_pattern;
+                for pattern in ignore_pattern {
+                    if pattern == path {
+                        return false;
+                    }
+
+                    if pattern.is_dir() && path.starts_with(pattern) {
+                        return false;
+                    }
+                }
+
+                true
+            })
```

Comment:
I thought we need to filter the walked paths? Not the passed in paths.

Reply:
Did you mean we should filter the pattern in https://github.com/Boshen/oxc/blob/main/crates/oxc_cli/src/walk.rs#L11C41-L18?

Reply:
Or after this line `.flat_map(|path| Walk::new(path).iter().collect::<Vec<_>>())`?

Reply:
`Walk::new(path).iter()` returns all the found path, so we need to filter on these.

Reply:
moved in 2f2cb78.

Reply:
I don't see how we are matching paths on these globs :-/

Reply:
Previous approach will expand all the matched paths and store them into `ignore_path`. The new approach use `Pattern` to check whether the path match the pattern.